Dockerfile

```diff
@@ -0,0 +1,56 @@
+# Docker tinsearch with deps
+#   - binaryen
+#   - wasm-pack
+#   - terser
+# For nightly rust toolset use build arg RUST_IMAGE=rustlang/rust:nightly-alpine
+#
+ARG WASM_REPO=https://github.com/mre/wasm-pack.git
+ARG WASM_BRANCH=first-class-bins
+ARG TINY_REPO=https://github.com/mre/tinysearch
+ARG TINY_BRANCH=master
+ARG RUST_IMAGE=rust:alpine
+
+
+FROM $RUST_IMAGE as binary-build
+
+ARG WASM_REPO
+ARG WASM_BRANCH
+ARG TINY_REPO
+ARG TINY_BRANCH
+ARG TINY_MAGIC
+
+
+WORKDIR /tmp
+
+RUN apk add --update --no-cache --virtual build-dependencies musl-dev openssl-dev gcc curl git npm gcc ca-certificates libc6-compat
+
+ENV env_var_name=$var_name
+
+ENV RUSTUP_HOME=/usr/local/rustup \
+    CARGO_HOME=/usr/local/cargo \
+    PATH=/usr/local/cargo/bin:$PATH \
+    WASM_REPO="$WASM_REPO" \
+    WASM_BRANCH="$WASM_BRANCH" \
+    TINY_REPO="$TINY_REPO" \
+    TINY_BRANCH="$TINY_BRANCH"
+
+RUN rustup target add asmjs-unknown-emscripten
+RUN rustup target add wasm32-unknown-emscripten
+
+RUN set -eux -o pipefail; \
+    ln -s /lib64/ld-linux-x86-64.so.2 /lib/ld64.so.1; \
+    npm install terser -g; \
+    curl -sL https://api.github.com/repos/WebAssembly/binaryen/releases/latest|grep tarball|awk '{print $2}'|sed 's/,//g'|xargs curl -sL |tar zxp ; \
+    cp -rp WebAssembly-binaryen*/* /usr/local/bin/.
+
+RUN time cargo install --force --git "$WASM_REPO" --branch "$WASM_BRANCH"
+
+RUN cd /tmp && git clone --branch "$TINY_BRANCH" "$TINY_REPO"
+RUN cd /tmp/tinysearch && if ! [[ -z $TINY_MAGIC ]]; then sed -i.bak bin/src/storage.rs -e "s/let mut filter = CuckooFilter::with_capacity(words.len() + 10);/let mut filter = CuckooFilter::with_capacity(words.len() + $TINY_MAGIC);/g";fi && cargo build --release && cp target/release/tinysearch $CARGO_HOME/bin && echo $TINY_MAGIC |tee /.tinymagic
+
+RUN wasm-pack --version
+RUN tinysearch --version
+
+RUN rm -rf /tmp/*; rm -rf /usr/local/cargo/{git,registry}
```

Comment:
I noticed that you were tagging this build as `FROM $RUST_IMAGE as binary-build`. If we had another `FROM alpine as final` here or something, we could avoid the cleanup here and have a very minimal build that only contains tinysearch and the optimization tools (binaryen and so on).
It's completely optional, though. Let me know if you like to change that before merge. Just a suggestion. 😉